manifests/mail_on_failure.pp

```diff
@@ -0,0 +1,43 @@
+# mail on systemd units failing
+#
+# This class manages a set of scripts that helps to
+# send out emails of failing units.
+# It prepares a unit template that can then be referenced
+# in units via OnFailure=status_email_USER@%n.service
+#
+# Email sending is handed over to sendmail and expected to
+# be allowed for user nobody.
+#
+# This is helpful to monitor failing systemd.timers in a cron-like
+# MAILTO style reporting.
+# See https://wiki.archlinux.org/index.php/Systemd/Timers#MAILTO
+# for background and implementation details.
+#
+# @param email_users
+#   Users to prepare a template unit for.
+#   By default this class prepares a template for root.
+#   Besides local users you can also define an email address.
+#   Due to the naming restrictionsof systemd units, the `@` in the template name
+#   will be replaced with `_AT_` and `ALIAS@DOMAIN` will get a unit called:
+#   `status_email_ALIAS_AT_DOMAIN@%n.service`
+#
+# @api public
+#
+class systemd::mail_on_failure (
+  Array[String[1]] $email_users = ['root'],
+) {
+  file {
+    '/usr/local/bin/systemd-email':
+      source => 'puppet:///modules/systemd/systemd-email',
```

Comment:
I'm a fan of `content => file('systemd/systemd-email')` which inlines it in the catalog. It also means a catalog compilation shows you if there's some reason the file isn't present (forgot git add, typo, etc).

Reply:
anything specific you would test for?

Reply:
I added a simple regexp on the content

Reply:
You can use `.with_content(/.../)` as a test. For fun, you can rename the file on disk and see the catalog compilation fail. That means your unit tests capture it rather than at runtime.

Another benefit is that if puppetserver is down, it can still use the cached catalog for a complete run.

It also used to be faster since it saves a separate HTTP request for a small file, but I don't know if that's still true since those tests were done a long time ago.